catroid/src/org/catrobat/catroid/physics/PhysicsObject.java

```diff
@@ -333,6 +328,19 @@ protected void setCollisionBits(short categoryBits, short maskBits) {
 			filter.maskBits = maskBits;
 			fixture.setFilterData(filter);
 		}
+
+		if (updateState) {
+			updateNonCollidingState();
+		}
+	}
+
+	private void updateNonCollidingState() {
+		if (body.getUserData() != null && body.getUserData() instanceof Sprite) {
+			Object look = ((Sprite) body.getUserData()).look;
+			if (look != null && look instanceof PhysicsLook) {
+				((PhysicsLook) look).setNonColliding(collisionMaskRecord == PhysicsWorld.MASK_NOCOLLISION);
+			}
```

Comment:
I don't like the whole method... :pouting_cat:
Not sure what this method does nor why the PhysicsObject has to know the sprite or look. Maybe you can do some magic and get rid of it...